iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -921,6 +951,18 @@ public void merge(Network other) {
         LOGGER.info("Merging of {} done in {} ms", id, System.currentTimeMillis() - start);
     }
 
+    private static Network createSubNetwork(NetworkImpl parent, Network original) {
+        Network sn = new SubNetworkImpl(parent, original.getId(), original.getOptionalName().orElse(null), original.getSourceFormat()).setCaseDate(original.getCaseDate());
+        new ArrayList<>(original.getExtensions())
```

Comment:
Why are extensions stored at the subNetwork level ? Is it to make the future split function easier to write ? Or to avoid the overwriting of potentially existing extensions on both merged networks ?

Reply:
Discussed over the phone: only the network extensions are transferred from the original network to the subnetwork in case of a merge. It is indeed to avoid the overwriting of potentially existing extensions on both networks.
The components extensions are not moved.